app/controllers/katello/api/v2/package_groups_controller.rb

```diff
@@ -3,6 +3,46 @@ class Api::V2::PackageGroupsController < Api::V2::ApiController
     apipie_concern_subst(:a_resource => N_("a package group"), :resource => "package_groups")
     include Katello::Concerns::Api::V2::RepositoryContentController
 
+    before_action :find_repository, :only => [:create, :destroy]
+
+    api :POST, "/package_group", N_("Create a package group")
+    param :repository_id, String, :required => true, :desc => N_("repository_id")
+    param :name, String, :required => true, :desc => N_("package group name.")
+    param :description, String, :desc => N_("package group description. Defaults to params[:name]")
+    param :user_visible, :bool, :desc => N_("set \"user_visible\" flag on package group. Defaults to true")
+    param :mandatory_package_names, Array, :desc => N_("mandatory package names to include in the package group")
+    param :optional_package_names, Array, :desc => N_("optional package names to include in the package group")
+    param :conditional_package_names, Array, :desc => N_("conditional package names to include in the package group")
+    param :default_package_names, Array, :desc => N_("default package names to include in the package group")
+
+    def create
+      fail Katello::Errors::InvalidRepositoryContent, _("Can only upload to Yum Repositories.") unless @repository.yum?
+      fail _("name not defined.") if params[:name].blank?
+
+      if params[:mandatory_package_names].empty? && params[:optional_package_names].empty? &&
+         params[:conditional_package_names].empty? && params[:default_package_names].empty?
+        fail HttpErrors::BadRequest, _("Must supply one of *_package_names param")
+      end
+
+      params[:description] = params[:name] if params[:description].empty?
+      params[:user_visible] = true if params[:user_visible].empty?
```

Comment:
This doesn't work if the user passes in   true  (the boolean) instead of 'true' the string.  You might use:
```
params[:user_visible] = ::Foreman::Cast.to_bool(params[:user_visible])
params[:user_visible] ||= true
```